### PR TITLE
Fixes #16057 - added dns_lookup template helper

### PR DESCRIPTION
**Pull request by lzap**
Also improved tests. Kindly ask reviewer to update our wiki page. Thanks!

### CODE REVIEW

File: lib/foreman/renderer.rb [hunk 1]
```diff
@@ -105,6 +105,18 @@ def indent(count)
       prefix + text.gsub(/\n/, "\n#{prefix}")
     end
 
+    def dns_lookup(name_or_ip)
+      resolver = Resolv::DNS.new
+      Timeout.timeout(Setting[:dns_conflict_timeout]) do
+        ipquery = IPAddr.new(name_or_ip) rescue false
```
Comment (dLobatog) [inline lib/foreman/renderer.rb]: @lzap Can we expect anything but a [IPAddr::Error](http://ruby-doc.org/stdlib-2.0.0/libdoc/ipaddr/rdoc/IPAddr/Error.html) here ?

File: test/lib/foreman/renderer_test.rb [hunk 2]
```diff
@@ -49,6 +41,38 @@ def setup_safemode_renderer
       assert_equal 'bar', tmpl
     end
 
+    test "foreman_server_fqdn helper method" do
```
Comment (dLobatog) [inline test/lib/foreman/renderer_test.rb:44]: Mind to move all the `send('setup_#{renderer_name}')` calls to a setup block in the .each loop?

File: lib/foreman/renderer.rb (hunk 1 above)
Comment (dLobatog) [reply lib/foreman/renderer.rb]: I tried with a variety of options, all of them fell under that class (most likely a subclass of it) of exception

**Comment by dLobatog** [comment]
@lzap Thanks, this would be useful. Two minor comments inline, also can you add tests for `dns_lookup` with invalid parameters? For instance a wrong IP and an invalid argument

File: lib/foreman/renderer.rb (hunk 1 above)
Comment (lzap) [reply lib/foreman/renderer.rb]: Sure.

File: test/lib/foreman/renderer_test.rb (hunk 2 above)
Comment (lzap) [reply test/lib/foreman/renderer_test.rb:44]: Yup.

Comment (lzap) [reply test/lib/foreman/renderer_test.rb:44]: That's can't be created actually, the setup block creates a closure around renderer_name and it's not set appopriately.

**Comment by lzap** [comment]
Rebased https://gist.github.com/8fd45cd74ef21d713c9c1ec78c6b02d7

**Comment by dLobatog** [comment]
[test] was aborted

**Comment by dLobatog** [comment]
Unrelated failure, merging

**Comment by dLobatog** [comment]
Merged as 1ffe8b2b472730be60d4126eaeb1e56253df2cb2, thanks @lzap!

**Pull request closed without merging**